### PR TITLE
Support negative shfit

### PR DESCRIPTION
**Pull request by splasky**
This commit is for support negative shift inside AMaCC for
consistent the result with gcc.

### CODE REVIEW

**Comment by splasky** [comment]
Solve #39

**Comment by jserv** [comment]
Mention "Close #39" in the end of Git commit message, so that #39 will be closed once this pull request is merged.

File: amacc.c [hunk 1]
```diff
@@ -615,8 +615,10 @@ void expr(int lev)
         case Gt:  next(); *++e = PSH; expr(Shl); *++e = GT;  ty = INT; break;
         case Le:  next(); *++e = PSH; expr(Shl); *++e = LE;  ty = INT; break;
         case Ge:  next(); *++e = PSH; expr(Shl); *++e = GE;  ty = INT; break;
-        case Shl: next(); *++e = PSH; expr(Add); *++e = SHL; ty = INT; break;
-        case Shr: next(); *++e = PSH; expr(Add); *++e = SHR; ty = INT; break;
+        case Shl: next(); *++e = PSH;if (tk == Sub){tk = Add;expr(Add);*++e = SHR;}
```
Comment (jserv) [inline amacc.c]: Follow the consistent coding style. There is space after `;`.

**Comment by jserv** [comment]
The line `Close #39` should be within Git commit messages rather than pull request discussions. Use `git rebase -i` to rework your commits and then force push.

File: amacc.c [hunk 2]
```diff
@@ -615,8 +615,10 @@ void expr(int lev)
         case Gt:  next(); *++e = PSH; expr(Shl); *++e = GT;  ty = INT; break;
         case Le:  next(); *++e = PSH; expr(Shl); *++e = LE;  ty = INT; break;
         case Ge:  next(); *++e = PSH; expr(Shl); *++e = GE;  ty = INT; break;
-        case Shl: next(); *++e = PSH; expr(Add); *++e = SHL; ty = INT; break;
-        case Shr: next(); *++e = PSH; expr(Add); *++e = SHR; ty = INT; break;
+        case Shl: next(); *++e = PSH; if (tk == Sub){tk = Add; expr(Add); *++e = SHR;}
+                                      else{expr(Add); *++e = SHL; }ty = INT; break;
```
Comment (jserv) [inline amacc.c]: Insert spaces before/after `{`.

Comment (jserv) [reply amacc.c]: Do follow the consistent style again.

**Pull request merged**